### ISSUE TITLE
1.8.9 support

### ISSUE THREAD

**Issue by mojang2555** [SkriptLang/Skript#847]
we need it

**Comment by Snow-Pyon** [comment]
It isn't a matter of needing it or not. You don't know how hard would be mantain 1.8.9 as it's **very** different from 1.9 and many things in the API are different between these versions.

@bensku is actually mantaining nearly 12 different versions, mantain another one which has many API changes would be a pain in the ass so unless you're gonna constantly do PR to keep the 1.8.9 support this wont happen.

**Comment by Syst3ms** [comment]
Ehhh, actually supporting 1.8 would (afaik) only need to rework some particle class using some reflection. But the thing is, do we really want to support a version that's something like 3 years old ?

**Comment by bensku** [comment]
After dropping 1.8 support, I have probably broken more than that. At least releases supporting 1.12.1 have broken a few expressions for 1.8 due to Spigot API changes.

Beyond that, I'm not interested in supporting 1.8. You could drop a bounty on this issue to get someone interested. To get me do this would require quite a lot of money, though :)

This Skript fork started to make Skript compatible with Minecraft 1.9. I don't know why you even expected it to continue working on 1.8

**Comment by Pikachu920** [comment]
@TheBentoBox I guess this can be closed when #1014 is merged? I'm not sure, it still isn't support however its probably the best this fork will get

**Comment by TheBentoBox** [comment]
Yeah agreed @Pikachu920. As Bensku said, this fork was started specifically for 1.9 so I think as long as it runs on 1.8.x without error, even if certain features are disabled, it can be considered "support" as far as this branch is concerned.

**Comment by Pikachu920** [comment]
it has been merged

**Comment by bensku** [comment]
1.8 is not supported until someone who wishes to fix bugs for it has push access. Maybe some day, but not yet.

**Comment by Pikachu920** [comment]
so i have push access now but i'm also not super interested in supporting 1.8 anymore. I will do my best to make it load on 1.8, but this will not be support. To clarify, by 1.8 I mean 1.8.8+

**Comment by Syst3ms** [comment]
tl;dr be happy that it even loads on 1.8

**Comment by Matocolotoe** [comment]
quite funny to use bStats as an argument but to forget **this** :)
![image](https://user-images.githubusercontent.com/42092549/57533108-3b80b200-733d-11e9-88d4-f04a08e45092.png)

**Comment by TheBentoBox** [comment]
By "supports 1.8" we mean actively supports it. Which they don't. There is a version of Spigot for 1.8 because of course there is but no, they do not provide active support for it nor does anyone in their right mind who's working on such a complex project.

Your graph is not a good argument. All you're proving to us is that 80% of people don't use 1.8. There are old versions of Skript that work fine on 1.8 and you're welcome to use those if you're using 1.8.

> wonder if somebody ever asks why 21% of worldwide servers stopped updating after 1.9

Probably because they're PvP servers or they're stuck in the past. There's no good reason otherwise; the game (PvE and gameplay-wise) has drastically improved since then in almost all areas. There's also an immense and toxic anti-post-1.8 circlejerk in the community so the fact that only 20% (and dropping) of servers are on 1.8 is a great sign. I can't wait to see that number eventually disappear.

I have half a mind to just close this now because this is likely never happening and, for the sake of the plugin's stability, it probably never should.

**Comment by Matocolotoe** [comment]
> Probably because they're PvP servers or they're stuck in the past. There's no good reason otherwise; the game (PvE and gameplay-wise) has drastically improved since then in almost all areas. There's also an immense and toxic anti-post-1.8 circlejerk in the community so the fact that only 20% (and dropping) of servers are on 1.8 is a great sign.

The fact is that there are many reasons not to update to 1.9+, though I understand why updating such a projet to 1.8.8 would be impossible due to where it is now.

> I can't wait to see that number eventually disappear.

Understandable due to all new things and (kind of) bug fixes in Minecraft now, but 1.8.x servers (such as Hypixel) and things such as ViaVersion won't ever disappear until 1.8 server owners just **all** quit the game or until Mojang releases a decent update which satisfies this community. The PvP community is a minor part of the whole Minecraft community but it can't be neglected atm

**Comment by bensku** [comment]
If part of Minecraft community wants new Skript releases for 1.8, they'll have to make them happen. Currently I don't think anyone in Skript team is interested in MC 1.8.

**Comment by Matocolotoe** [comment]
I got 2 dislikes for no reason lmao

**Comment by Matocolotoe** [comment]
> I can't wait to see that number eventually disappear.

I guess it will when Mojang releases an update which satisfies this community.
Even Hypixel thinks about dropping support for 1.9+...

**Comment by Matocolotoe** [comment]
So I actually made the plugin load successfully in 1.8.8 (at least without errors on startup, doesn't include errors I could have in advanced scripts) and I had to :
- seperate 1.9 aliases from 1.8- ones and write `minecraft version = 1.9 or newer` above those
- remove `OcelotData.java`
- remove 1.9 boat stuff from `BoatData.java`
- remove all 1.9+ biomes from `BiomeMapUtil.java`
- add a `1.8.json` file in `materials` folder
Nearly impossible unless people fork the code and do this by themselves, so I think this issue should be closed.

**Comment by Pikachu920** [comment]
why? it would be possible to support 1.8

**Comment by Matocolotoe** [comment]
Well that's really weird since the biome thing doesn't break with 1.9 even if it has 1.12 biomes in it.

**Comment by bensku** [comment]
Biome mappings are a bit weird. Currently, they map names used by MC 1.9-1.12 to names used by MC 1.13. What they *used* to do was map names used by MC 1.9-1.12 to names used by 1.8. The mapping system does not support doing both, thus biomes are not working on 1.8 nowadays.

**Comment by Matocolotoe** [comment]
Oh, okay, thanks for the info !
Actually making the plugin load on 1.8 without errors took even more work, I literally spent a whole day on this to figure out what to remove and what to change not to throw any error...

**Comment by Matocolotoe** [comment]
If anyone wants to check it out : https://github.com/Matocolotoe/Skript-1.8

**Comment by jaylawl** [comment]
https://minecraft-statistic.net/en/global_statistic.html

According to this, 1.8.8 servers are down to 7.3 %.

I'm taking a bit of freedom to speculate here; i think most of these servers do it for monetary reasons and usually have their own (paid) programmers for custom stuff anyway.

**Comment by ShaneBeee** [comment]
> https://minecraft-statistic.net/en/global_statistic.html
> 
> According to this, 1.8.8 servers are down to 7.3 %.
> 1.8.9 = 0.1 %
> 1.7.10 = 4.2 %
> 
> I'm taking a bit of freedom to speculate here; i think most of these servers do it for monetary reasons and usually have their own (paid) programmers for custom stuff anyway.

I think that websites stats aren't showing proper numbers. For instance, it says 1759 servers are running 1.14.4, yet bStats shows 25,000 are running 1.14.4, so by comparison, that website is showing a very small amount of servers that are actually running.
My assumption is its only showing servers registered to that website. 

Not that I am endorsing 1.8.8, because I think its time everyone moves on, but 1.8.8 is the second most used server version according to bStats at 16% (global stats)
<img width="525" alt="Screen Shot 2019-07-31 at 11 25 09 AM" src="https://user-images.githubusercontent.com/20008482/62237479-e1185000-b385-11e9-951e-ea9b0ee7d694.png">


And if we look at bStats for Skript:
<img width="525" alt="Screen Shot 2019-07-31 at 11 26 17 AM" src="https://user-images.githubusercontent.com/20008482/62237592-1755cf80-b386-11e9-9c56-611cb088695b.png">
We can see that just shy of 10% of servers using Skript, are running 1.8.8, and about 68.5% of servers running Skript are running a more recent version of Spigot/MC (68.5 is a total of all the servers running 1.12.2, 1.13.2 and 1.14.4)

**Comment by Matocolotoe** [comment]
Still, it's nicer to help people looking for Skript in 1.8 than just saying "u part of the 10% update to 1.14.4 or use dev36"

**Comment by TPGamesNL** [comment]
Since we're going to drop support for 1.9-1.12 soon, I hardly doubt we'll be adding support back for an even older version soon. There's also [the Skript-1.8 fork](https://github.com/Matocolotoe/Skript-1.8) now, so I'm closing this issue
